src/OrleansRuntime/Core/Dispatcher.cs

```diff
@@ -564,6 +564,7 @@ internal void SendResponse(Message request, Response response)
         {
             // create the response
             var message = request.CreateResponseMessage();
+            if(!request.DisposeScheduled) request.Dispose();
```

Comment:
you sure its the right condition? !DisposeScheduled then dispose? There is no other logic for auto dispose, so for me DisposeScheduled means - if true, do dispose.
Not sure why you need this flag at all. But even if yes, not sure about the true/false right meaning.